polycommit/src/sonic_pc/gadgets/verifier_key/prepared_verifier_key.rs

```diff
@@ -67,77 +60,9 @@ where
-
-            let desired_bound_value = bound.get_value().unwrap_or_default();
-
-            for (i, (_, this_bound, _)) in degree_bounds_and_prepared_neg_powers_of_h.iter().enumerate() {
-                if this_bound.get_value().unwrap_or_default().eq(&desired_bound_value) {
-                    pir_vector[i] = true;
-                    break;
-                }
-            }
-
-            let mut pir_vector_gadgets = Vec::with_capacity(pir_vector.len());
-            for (i, bit) in pir_vector.iter().enumerate() {
-                pir_vector_gadgets.push(Boolean::alloc(cs.ns(|| format!("alloc_pir_{}", i)), || Ok(bit))?);
-            }
-
-            // Sum of the PIR values are equal to one
-            let mut sum = FpGadget::<<BaseCurve as PairingEngine>::Fr>::zero(cs.ns(|| "zero"))?;
-            let one = FpGadget::<<BaseCurve as PairingEngine>::Fr>::one(cs.ns(|| "one"))?;
-            for (i, pir_gadget) in pir_vector_gadgets.iter().enumerate() {
-                let temp = FpGadget::<<BaseCurve as PairingEngine>::Fr>::from_boolean(
-                    cs.ns(|| format!("from_boolean_{}", i)),
-                    pir_gadget.clone(),
-                )?;
-
-                sum = sum.add(cs.ns(|| format!("sum_add_pir{}", i)), &temp)?;
-            }
-            sum.enforce_equal(cs.ns(|| "sum_enforce_equal"), &one)?;
-
-            // PIR the value
-            let zero_bound = FpGadget::<<BaseCurve as PairingEngine>::Fr>::zero(cs.ns(|| "zero_bound"))?;
-            let zero_shift_power = PG::G2PreparedGadget::zero(cs.ns(|| "zero_shift_power"))?;
-            let mut sum_bound = zero_bound.clone();
-            let mut shift_powers_to_be_summed = Vec::new();
-
-            for (i, (pir_gadget, (_, degree, shift_power))) in pir_vector_gadgets
-                .iter()
-                .zip(degree_bounds_and_prepared_neg_powers_of_h.iter())
-                .enumerate()
-            {
-                let found_bound = FpGadget::<<BaseCurve as PairingEngine>::Fr>::conditionally_select(
-                    cs.ns(|| format!("found_bound_cond_select{}", i)),
-                    pir_gadget,
-                    degree,
-                    &zero_bound,
-                )?;
-                sum_bound.add_in_place(cs.ns(|| format!("update sum_bound{}", i)), &found_bound)?;
-
-                let found_shift_power = PG::G2PreparedGadget::conditionally_select(
-                    cs.ns(|| format!("found_shift_conditionally_select{}", i)),
-                    pir_gadget,
-                    shift_power,
-                    &zero_shift_power,
-                )?;
-
-                shift_powers_to_be_summed.push(found_shift_power);
-            }
-
-            sum_bound.enforce_equal(cs.ns(|| "found_bound_enforce_equal"), &bound)?;
-
-            let sum_shift_power =
-                PG::G2PreparedGadget::sum(cs.ns(|| "sum the shift powers for PIR"), &shift_powers_to_be_summed)?;
-            Ok(sum_shift_power)
-        }
```

Comment:
This was removed because, in the context of the Marlin verifier, selecting the appropriate prepared key is much more expensive than selecting the unprepared element and then preparing it (we only access two elements, each once)